### PR TITLE
peer: use generic lru

### PR DESCRIPTION
**Pull request by tuxcanfly**
This PR updates `sentNonces` and `knownInventory` to use the more generic LRU from Decred.  This cuts down the need for maintaining two instances of almost same code i.e `mruInventoryMap` and `mruSentNonces`.

### CODE REVIEW

**Comment by jcvernaleo** [comment]
@dajohi @davecgh do either of you see any problem important the code from decred in this way?  It seems like a good way to go but wanted to give you two a chance to chime in.

File: peer/peer.go [hunk 1]
```diff
@@ -450,7 +451,7 @@ type Peer struct {
 
 	wireEncoding wire.MessageEncoding
 
-	knownInventory     *mruInventoryMap
+	knownInventory     lru.Cache
```
Comment (onyb) [inline peer/peer.go:454]: Could this have been a pointer (i.e. *lru.Cache)?

File: go.sum [hunk 2]
```diff
@@ -24,6 +24,9 @@ github.com/btcsuite/winsvc v1.0.0/go.mod h1:jsenWakMcC0zFBFurPLEAyrnc/teJEM1O46f
 github.com/davecgh/go-spew v0.0.0-20171005155431-ecdeabc65495/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
 github.com/davecgh/go-spew v1.1.1 h1:vj9j/u1bqnvCEfJOwUhtlOARqs3+rkHYY13jYWTU97c=
 github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSsI+c5H38=
+github.com/decred/dcrd v1.3.0 h1:EEXm7BdiROfazDtuFsOu9mfotnyy00bgCuVwUqaszFo=
```
Comment (davecgh) [inline go.sum:27]: This doesn't look right to me.  The lru module has no dependencies.  I expect this needs a `go mod tidy`.

**Comment by davecgh** [comment]
I don't see anything wrong with using dcrd modules.  Modules are stable and can be used by callers as desired.

File: go.sum (hunk 2 above)
Comment (tuxcanfly) [reply go.sum]: Thanks, fixed.

File: peer/peer.go (hunk 1 above)
Comment (tuxcanfly) [reply peer/peer.go:454]: Not really useful as it's not being passed around anywhere.

**Comment by jcvernaleo** [comment]
@davecgh did you have any objections of than the go mod tidy (which was done)?  If not I'm going to merge this one.

**Comment by jcvernaleo** [comment]
Since there seemed to be no objections and this is important, merging it in now.

**Comment by jcvernaleo** [comment]
Actually, @tuxcanfly could you squash this to one commit before I merge?

**Comment by davecgh** [comment]
I didn't test it, but it reads well enough.

**Comment by jakesyl** [comment]
Will give it a quick test

**Comment by tuxcanfly** [comment]
Squashed :+1:

**Comment by tuxcanfly** [comment]
Just a sec, borrowing some updates from @dajohi

**Comment by jcvernaleo** [comment]
@tuxcanfly just squash it when you are ready for final review/merge/

File: netsync/manager.go [hunk 3]
```diff
@@ -147,6 +147,25 @@ type peerSyncState struct {
 	requestedBlocks map[chainhash.Hash]struct{}
 }
 
+// limitAdd is a helper function for maps that require a maximum limit by
```
Comment (tuxcanfly) [inline netsync/manager.go:150]: Does anyone object to this func name? It's called `addAndLimitMap` in @dajohi code but I thought it was too verbose.

**Comment by tuxcanfly** [comment]
Squashed again.

File: netsync/manager.go (hunk 3 above)
Comment (jcvernaleo) [reply netsync/manager.go:150]: no objection from me

**Pull request merged**